frontend/src/components/pages/topics/Topic.List.tsx

```diff
@@ -112,24 +112,41 @@ class TopicList extends PageComponent {
     void api.refreshClusterHealth();
   }
 
-  isFilterMatch(filter: string, item: Topic): boolean {
-    try {
-      const quickSearchRegExp = new RegExp(filter, 'i');
-      return Boolean(item.topicName.match(quickSearchRegExp));
-    } catch (e) {
-      console.warn('Invalid expression');
-      return item.topicName.toLowerCase().includes(filter.toLowerCase());
-    }
-  }
-
   @computed get topics() {
     let topics = api.topics ?? [];
     if (uiSettings.topicList.hideInternalTopics) {
       topics = topics.filter((x) => !x.isInternal && !x.topicName.startsWith('_'));
     }
+
+    if (uiSettings.topicList.quickSearch) {
+      try {
+        const quickSearchRegExp = new RegExp(uiSettings.topicList.quickSearch, 'i');
+        topics = topics.filter(
+          (topic) => Boolean(topic.topicName.match(quickSearchRegExp))
+        );
+      } catch (e) {
+        console.warn('Invalid expression');
+        const searchLower = uiSettings.topicList.quickSearch.toLowerCase();
+        topics = topics.filter(
+          (topic) => topic.topicName.toLowerCase().includes(searchLower)
+        );
+      }
+    }
+
     return topics;
   }
 
+  SearchBar = observer(() => {
+    return (
+      <SearchField
+        width="350px"
+        placeholderText="Enter search term/regex"
+        searchText={uiSettings.topicList.quickSearch}
+        setSearchText={(x) => (uiSettings.topicList.quickSearch = x)}
+      />
+    );
+  });
```

Comment:
What's the difference between what we had previously and this change? Is it required for MobX state management? If so, can we leave a comment?

Reply:
The other search was causing infinite loops and re-renders, so I had to replace it. I'll need to backport the functionality that shows numbers about filtered items.